### DISCUSSION TITLE
New to TS/JS. Trying to run example.ts

### DISCUSSION THREAD

**Discussion by ColtonIdle** [parnic/node-screenlogic#78] [Q&A] [answered]
I'm trying to run example.js

I cloned the repo. Cd'd into it. `npm install -g typescript`. then tsc example.ts   

but im hit with

```
Found 266 errors in 6 files.

Errors  Files
     3  example.ts:7
   249  index.ts:4
     3  messages/OutgoingMessages.ts:14
     1  messages/SLGatewayDataMessage.ts:8
     7  messages/SLMessage.ts:3
     2  utils/PasswordEncoder.ts:56
```

am i missing something? i am mostly a mobile/kotlin dev so im out of my depth here 😅

**Comment by parnic** [comment, marked as answer]
You should:
* Clone the repo
* Cd into it
* Run: `npm install` to get dependencies installed locally, not a global version of a dependency
* Run ~~`npm build`~~ `npm run build` if you want to re-compile the typescript, but this is not required as the compiled files are already in the "dist" folder
* `node dist/example.js` to run the example script

**Reply by ColtonIdle** [reply]
```
➜ node-screenlogic-main npm build    
Unknown command: "build"

Did you mean this?
  npm run build # run the "build" package script

To see a list of supported npm commands, run:
  npm help
```

**Reply by parnic** [reply]
Sorry, yes, `npm run build`. Typing from memory :)

**Reply by ColtonIdle** [reply]
cool. sorry for the noob questions and appreciate the hand holding!

thanks for teaching (once again!)

**Reply by ColtonIdle** [reply]
excellent! Im able to connect now! woot! now to try to add this to my project. im assuming since this isn't distributed through "npm" I just need to include this in the root of my project

**Reply by parnic** [reply]
It is on npm. 

https://www.npmjs.com/package/node-screenlogic

**Reply by ColtonIdle** [reply]
Oh snap. Sorry. Typically on android/kotlin repos it'll say the maven coordinates for integrating a library. So maybe I missed this on the readme when I read through it
